### ISSUE TITLE
there may exist a code bug

### ISSUE THREAD

**Issue by chenxianbo** [cvg/LightGlue#43]
The code of m1 may exist a bug, the shape of attn10 is 'b h j i', so  the shape of attn10.transpose(-2, -1) is 'b h i j', can you help check that？

           “qk0, qk1 = qk0 * self.scale**0.5, qk1 * self.scale**0.5
            sim = torch.einsum('b h i d, b h j d -> b h i j', qk0, qk1)
            attn01 = F.softmax(sim, dim=-1)
            attn10 = F.softmax(sim.transpose(-2, -1).contiguous(), dim=-1)
            m0 = torch.einsum('bhij, bhjd -> bhid', attn01, v1)
            m1 = torch.einsum('bhji, bhjd -> bhid', attn10.transpose(-2, -1), v0)”
